apps/sms/views/conversation/js/conversation.js

```diff
@@ -2973,52 +2837,51 @@ var ConversationView = {
     return defer.promise;
   },
 
-  discardDraft: function conv_discardDraft() {
-    // If we were tracking a draft properly, update the Drafts object entry.
-    if (!this.draft) {
-      return;
-    }
+  updateDraft() {
+    var conversation = this.activeThread;
 
-    Drafts.delete(this.draft).store();
+    // Need to assimilate recipients in order to check if any entered.
+    this.assimilateRecipients();
 
-    this.draft = null;
-  },
+    // If message content is not empty and/or we have _valid_ recipients in
+    // NewMessage view.
+    var hasDraftToSave = !Compose.isEmpty() ||
+      (!conversation && this.recipients.numbers.length);
 
-  /**
-   * Saves currently unsent message content and/or recipients into a Draft
-   * object. Preserves currently active draft if specified. Draft preservation
-   * is intended to keep this.draft populated with the currently active draft
-   * when app is hidden, so that when app becomes visible again, it knows there
-   * is a draft to continue to keep track of.
-   *
-   * @param {boolean?} preserveDraft Optional boolean parameter that indicates
-   * whether or not we should preserve draft.
-   */
-  saveDraft: function conv_saveDraft(preserveDraft) {
-    var thread = this.activeThread;
+    // Check if conversation has been just removed, can happen when user selects
+    // all messages in Conversation view and removes all of them, in this case
+    // we don't want to loose draft.
+    var isConversationRemoved = conversation && !conversation.messages.size;
+
+    // If we have active draft, but user cleared it up or removed conversation
+    // this draft belonged to, we should discard it.
+    var isDraftDiscarded = false;
+    if (this.draft && (!hasDraftToSave || isConversationRemoved)) {
+      Drafts.delete(this.draft);
 
-    // Do we need to save participants for thread draft?
-    var recipients = thread ? thread.participants : this.recipients.numbers;
+      this.draft = null;
+
+      isDraftDiscarded = true;
+    }
+
+    // If we don't have any content to save to draft or existing draft has not
+    // been edited yet, we should not do anything.
+    if (!hasDraftToSave || (this.draft && !this.draft.isEdited)) {
+      return isDraftDiscarded ? Drafts.store() : Promise.resolve();
+    }
 
-    var draft = new Draft({
+    // Otherwise we should save new draft.
+    this.draft = new Draft({
       id: this.draft && this.draft.id,
-      threadId: thread && thread.id,
-      recipients: recipients,
+      threadId: !isConversationRemoved ? conversation && conversation.id : null,
+      recipients: conversation ?
+        conversation.participants : this.recipients.numbers,
       content: Compose.getContent(),
-      subject: Compose.getSubject(),
+      subject: Compose.isSubjectVisible ? Compose.getSubject() : null,
```

Comment:
actually, I think that `Compose.getSubject()` should not return the text if it's not visible. Which would fix the aforementioned STR too. And actually that was the case before bug 1067228.

This should likely be a separate bug so that we can uplifted to 2.2.
What do you think ?